features/features.go

```diff
@@ -522,6 +522,14 @@ var (
 				enableForClusterProfile(SelfManaged, configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade, configv1.Default).
 				mustRegister()
 
+	FeatureGateNewOLMCatalogdMetas = newFeatureGate("NewOLMCatalogdMetas").
```

Comment:
Should we include `APIV1` in the name in case there is ever an `/api/v2/metas` endpoint behind a different feature gate?
```suggestion
	FeatureGateNewOLMCatalogdMetas = newFeatureGate("NewOLMCatalogdAPIV1Metas").
```